src/metabase/driver/common.clj

```diff
@@ -229,7 +230,12 @@
 (def auth-provider-options
   "Options for using an auth provider instead of a literal password."
   [{:name "use-auth-provider"
-    :type :section
+    :type :checked-section
+    :check (fn []
+             (and
+               ;; Managed Identities only make sense if Metabase is in the same cloud as the DW
+              (not (premium-features/is-hosted?))
```

Comment:
I thought the feature will be available also in cloud deployments.

Reply:
The idea is that PaaS providers have ways to authenticate with their dbs when running metabase in the same cloud. Until our hosting allows VPC this will not work for them.

 With Azure, if you run metabase on a azure, you can use managed identity, with AWS if you run metabase in the same account as the DB you can use their version, etc..